### PR TITLE
fix(ivy): platform module bootstrap does not resolve resources 

### PR DESCRIPTION
**Pull request by devversion**
* See individual commits.

### CODE REVIEW

File: packages/core/src/application_ref.ts [hunk 1]
```diff
@@ -49,7 +51,30 @@ export function compileNgModuleFactory__POST_R3__<M>(
...
     moduleType: Type<M>): Promise<NgModuleFactory<M>> {
   ngDevMode && assertNgModuleType(moduleType);
-  return Promise.resolve(new R3NgModuleFactory(moduleType));
+  const moduleFactory = new R3NgModuleFactory(moduleType);
+
+  if (isComponentResourceResolutionQueueEmpty()) {
+    return Promise.resolve(moduleFactory);
+  }
+
+  const compilerOptions = injector.get(COMPILER_OPTIONS, []).concat(options);
```
Comment (alxhub) [inline packages/core/src/application_ref.ts]: I'm confused here... `options` is not an array type, how can it be concat'd?

**Review by gkalpak — APPROVED** [review]
LGTM for `docs-infra` :grin:

File: packages/core/src/application_ref.ts (hunk 1 above)
Comment (devversion) [reply packages/core/src/application_ref.ts]: The `COMPILER_OPTIONS` injection token declares an array of compiler options. There can be multiple compiler options from the chained platforms. So this is guaranteed to be an array where we just concat the explicitly specified compiler options from the developer.

e.g.

```ts
platform().bootstrapModule(MyMod, {
  providers: [    /* Overwrite resource loader impl. */  ]
});
```

**Review by alxhub — APPROVED** [review]

**Comment by kara** [comment]
[presubmit](http://test/OCL:238048116:BASE:238048183:1552412580795:822512a6)

**Comment by kara** [comment]
@devversion Could you create a separate PR for patch? Not merging cleanly

**Pull request closed without merging**